### PR TITLE
Add zero-ing of Azure2021 filter function

### PR DESCRIPTION
**Pull request by 16fb**
## Summary
* Zeros the resultant azure2021 trace to the smallest `start_timestamp` value.

## Implementation Notes :hammer_and_pick:

* Add in `zero_timestamps` function in filter.py
* After keeping functions present in both sampled and original trace, subtracts the smallest `start_timestamp` value from both `start_timestamp` and `end_timestamp` columns.

## External Dependencies :four_leaf_clover:

none (N/A)

## Breaking API Changes :warning:

none (N/A)

### CODE REVIEW

File: sampler/tests/filter_test.py [hunk 1]
```diff
@@ -154,3 +154,56 @@ def test_same_app_different_functions(tmp_path):
...
+
+    # Read and compare output filtered_DF
+    filtered_df_path = out_dir / "SampledAzure2021.csv"
+    filtered_df = pd.read_csv(filtered_df_path)
+
+    expected_filtered_df = pd.DataFrame(
+        {
+            "app":           ["ac",  "ac"],
+            "func":          ["fa",  "fb"],
+            "end_timestamp": [20.0,  40.0],
```
Comment (leokondrashov) [inline sampler/tests/filter_test.py]: can you add the test that would end up with non-zero timestamp for the first invocation as we discussed? this should subtract 60s (the start_time), not necesserily the minimum of start timestamps

File: docs/sampler.md [hunk 2]
```diff
@@ -175,7 +175,7 @@ in [`plotTimeline/analysis.ipynb`](/tools/plotTimeline/analysis.ipynb)
 The sampling process of Azure2021 trace follows 3 steps: 
 1. Pre-process the original trace (`preprocess2021`) -> obtain clean trace in azure2019 format.
 2. Run the sampler on cleaned trace (`sampler`) -> obtain down-sampled trace, which contains a list of functions to keep in azure2019 format.
-3. Filter the original trace for functions in list (`filter`) -> obtain down-sampled trace in azure2021 format (per-invocation format).
+3. Filter the original trace for functions in list (`filter`) -> obtain down-sampled and zero-ed trace in azure2021 format (per-invocation format).
```
Comment (leokondrashov) [inline docs/sampler.md]: i would call it timestamp normalization, not zero-ed. that might be misunderstood

File: sampler/tests/filter_test.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def test_azure2021_filter(tmp_path):
         {
             "app":           ["aa", "ab", "ac", "ac"],
             "func":          ["fa", "fb", "fd", "fd"],
-            "end_timestamp": [1.00, 10.0, 100.0, 300.0],
+            "end_timestamp": [ 1.0, 10.0, 100.0, 300.0], # All zeroed to start_time's Timedelta of 0
```
Comment (leokondrashov) [inline sampler/tests/filter_test.py]: so, unchanged, right? i would just remove the comment, not to confuse anyone

File: sampler/tests/filter_test.py [hunk 4]
```diff
@@ -154,3 +154,56 @@ def test_same_app_different_functions(tmp_path):
...
 
+# - Test that output DF zeros to start_time
+def test_zero_to_start_time(tmp_path):
+
+    # Original DF
+    og_df = pd.DataFrame(
+        {
+            "app":           ["ac",  "ac"],
+            "func":          ["fa",  "fb"],
+            "end_timestamp": [80.0, 100.0], # 4 Minutes, 60-300 seconds
```
Comment (leokondrashov) [inline sampler/tests/filter_test.py]: the comment is weird, i guess it is copypasted from the other test with more requests

**Review by leokondrashov — COMMENTED** [review]
looks good, a few rough small nitpicks here and there and good to go

File: sampler/tests/filter_test.py (hunk 1 above)
Comment (16fb) [reply sampler/tests/filter_test.py]: Oops mb, didn't notice that the 2nd timestamp started at 0.

**Comment by 16fb** [comment]
Have incorporated the fixes you mentioned.

The Link check is returning 403 forbidden for a link to https://www.ibm.com/docs/en/cloud-private/3.1.2?topic=networking-kubernetes-network-model .

I think lychee has an .ignore/exclude file, should the link/domain be added to that?

**Comment by leokondrashov** [comment]
if you follow the link, there is a banner at the top that there is newer version of the article, maybe they are responding with 403 because of that. can you try updating the link? if it doesn't help, try to find another article that covers the content. for sure there are some other posts about the pod cidrs that would be responding normally.

**Review by leokondrashov — DISMISSED** [review]
fix the link, otherwise ready to merge

**Comment by 16fb** [comment]
Changed CIDR explanation link to https://www.redhat.com/en/blog/cni-kubernetes

**Review by leokondrashov — APPROVED** [review]
